x/concentrated-liquidity/internal/math/math.go

```diff
@@ -85,8 +99,9 @@ func CalcAmount1Delta(liq, sqrtPriceA, sqrtPriceB sdk.Dec, roundUp bool) sdk.Dec
 	// additionally, without rounding, there exists cases where the swapState.amountSpecifiedRemaining.GT(sdk.ZeroDec()) for loop within
 	// the CalcOut/In functions never actually reach zero due to dust that would have never gotten counted towards the amount (numbers after the 10^6 place)
 	if roundUp {
-		return liq.Mul(diff).Ceil()
+		return liq.Mul(diff).RoundInt().ToDec()
 	}
```

Comment:
Note to reviewer: Come to a conclusion on what the proper rounding strategy is (ceil does not act as I would expect it to)

Reply:
Ditto here! Would it be possible to make an issue between the tradeoffs? Thanks!

Reply:
Reverted https://github.com/osmosis-labs/osmosis/pull/4051/commits/6958a3d47f21635d1aa05858757f7ab18e3ae9b9